src/query/service/src/interpreters/interpreter.rs

```diff
@@ -85,7 +85,10 @@ pub trait Interpreter: Sync + Send {
     #[fastrace::trace]
     async fn execute(&self, ctx: Arc<QueryContext>) -> Result<SendableDataBlockStream> {
         log_query_start(&ctx);
-        match self.execute_inner(ctx.clone()).await {
+        let res = self.execute_inner(ctx.clone()).await;
```

Comment:
I'm not sure if the pipeline finishes executing here. It looks like `execute_inner()` returns a `SendableDataBlockStream` and needs to be polled later. Maybe we should call `ctx.drop_m_cte_temp_table()` with `pipeline.set_on_finished()`.